extra/interface/ap/src/main/java/org/spongepowered/configurate/interfaces/processor/ConfigImplementationGenerator.java

```diff
@@ -0,0 +1,169 @@
+            if (element.isDefault()) {
+                if (excluded) {
+                    // no need to handle them
+                    continue;
+                }
+                this.processor.info("Overriding implementation for %s as it's not excluded", element);
+            } else if (excluded) {
+                throw new IllegalStateException(String.format(
+                    Locale.ROOT,
+                    "Cannot make config due to method %s, which is a method excluded method that has no implementation!",
+                    element
+                ));
+            }
+
+            // all methods are either setters or getters past this point
+
+            final List<? extends VariableElement> parameters = element.getParameters();
+            if (parameters.size() > 1) {
+                throw new IllegalStateException("Setters cannot have more than one parameter! Method: " + element);
+            }
+
+            final String simpleName = element.getSimpleName().toString();
+            TypeMirror nodeType = element.getReturnType();
+
+            if (parameters.size() == 1) {
+                // setter
+                final VariableElement parameter = parameters.get(0);
+
+                final MethodSpec.Builder method = MethodSpec.overriding(element)
+                    .addStatement(
+                        "this.$N = $N",
+                        element.getSimpleName(),
+                        parameter.getSimpleName()
+                    );
+
+                // if it's not void
+                if (!MoreTypes.isTypeOf(Void.TYPE, nodeType)) {
+                    // the return type can be a parent type of parameter, but it has to be assignable
+                    if (!this.processor.typeUtils.isAssignable(parameter.asType(), nodeType)) {
+                        throw new IllegalStateException(String.format(
+                            "Cannot create a setter with return type %s for argument type %s. Method: %s",
+                            nodeType,
+                            parameter.asType(),
+                            element
+                        ));
+                    }
+                    method.addStatement("return this.$N", element.getSimpleName());
+                }
+
+                spec.add(simpleName + "#" + parameter.getSimpleName().toString(), method);
+                nodeType = parameter.asType();
+            } else {
+                // getter
+                spec.add(
+                    simpleName,
+                    MethodSpec.overriding(element)
+                        .addStatement("return $N", element.getSimpleName())
+                );
+            }
+
+            final FieldSpec.Builder fieldSpec = FieldSpec.builder(TypeName.get(nodeType), simpleName, Modifier.PRIVATE);
+
+            //todo add tests for hidden in both ap and interfaces and defaults in interfaces
+            AnnotationDefaults.process(element, nodeType, fieldSpec);
```

Comment:
I think it makes more sense to just copy all annotations, and error if they're not applicable for `TYPE_USE` or `FIELD` (you can use iirc `Elements` to look up types by name for this sort of info)